File: helium/src/main/groovy/com/stanfy/helium/handler/codegen/objectivec/file/ObjCClass.java
package com.stanfy.helium.handler.codegen.objectivec.file;

/**
 * Created by ptaykalo on 8/17/14.
 * Simple object structure that holds information about objectiveC class
 */
public class ObjCClass {

  /*
  Class name is unique in the project, since objC still doesn't have normal packages or namespaces
   */
  private String name;
  /*
  Link to class implementation part.
  This part lives in the ObjCImplementationFile
   */
  private ObjCClassImplementation implementation;
    /*
  Link to class definition part.
  This part lives in the ObjCCDefinitionFile
   */
  private ObjCClassDefinition definition;

  public ObjCClass(final String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public ObjCClassImplementation getImplementation() {
    return implementation;
  }
  public void setImplementation(final ObjCClassImplementation implementation) {
    this.implementation = implementation;
  }

  public ObjCClassDefinition getDefinition() {
    return definition;
  }
  public void setDefinition(final ObjCClassDefinition definition) {
    this.definition = definition;
  }
}

File: codegen/objc/objc-entities/src/main/kotlin/com/stanfy/helium/handler/codegen/objectivec/entity/classtree/ObjCClass.kt
package com.stanfy.helium.handler.codegen.objectivec.entity.classtree;

public interface  ObjCClassType {
  val name: String
}
/**
 * Created by ptaykalo on 8/17/14.
 * Simple object structure that holds information about objectiveC class
 */
public class ObjCClass(override val name: String, val definition: ObjCClassInterface,
                       val implementation: ObjCClassImplementation) : ObjCClassType {

  constructor(name:String):this(name, ObjCClassInterface(name), ObjCClassImplementation(name))
  public var forwardDeclarations = hashSetOf<String>()
    private set

  /**
   * Adds external class declaration string. This one should be transformed to "@class |externalClass|" in the eneratir
   */
  public fun addForwardDeclaration(externalClass: String) {
    forwardDeclarations.add(externalClass)
  }

}


public class ObjCPregeneratedClass(override val name:String, val header:String?, val implementation:String?) :ObjCClassType {

}

